generator/MarshalGen.cs

```diff
@@ -60,5 +60,22 @@ public override string FromNative (string var)
 			return String.Format (from_fmt, var);
 		}
 	}
+
+	public class StringMarshalGen : MarshalGen, IManualMarshaler
+	{
+		public StringMarshalGen (string ctype, string type, string mtype, string call_fmt, string from_fmt) : base (ctype, type, mtype, call_fmt, from_fmt, true)
+		{
+		}
+
+		public string AllocNative (string managed_var)
+		{
+			return CallByName (managed_var);
+		}
+
+		public string ReleaseNative (string native_var)
+		{
+			return $"GLib.Marshaller.Free ({native_var})";
```

Comment:
Does gtk-sharp depend already in a high enough version of mono to support interpolated strings?